### ISSUE TITLE
Cannot send email

### ISSUE THREAD

**Issue by poudenes** [hassio-addons/addon-bitwarden#20]
# Problem/Motivation

When I add all the mailserver information. Try to invite a person I get a error in de mailserver log:

ssl.SSLError: [SSL: WRONG_VERSION_NUMBER] wrong version number

Mailserver is configured to use TLS:

# TLS parameters
smtpd_tls_cert_file = /etc/postfix/smtpd.cert
smtpd_tls_key_file = /etc/postfix/smtpd.key
smtpd_use_tls = yes
smtpd_tls_session_cache_database = btree:${data_directory}/smtpd_scache
smtp_tls_session_cache_database = btree:${data_directory}/smtp_scache

smtp_tls_security_level = may
smtpd_tls_mandatory_protocols = !SSLv2, !SSLv3
smtpd_tls_protocols = !SSLv2,!SSLv3
smtp_tls_protocols = !SSLv2,!SSLv3

Everything what I try in config of Bitwarden its not sending emails.

It's not the mailserver because all other normal mails will receive and send.

**Comment by woody4165** [comment]
Hi

I'm getting too issue to send email, also the test email.
The error I get is
![Schermata 2020-04-24 alle 17 57 58](https://user-images.githubusercontent.com/6501444/80232959-05c00080-8656-11ea-8e46-59c0310ad57d.png)

I saw a Bitwarden Github issue when it works with TLS, but I've defined Gmail as SMTP server with SSL, but without TLS.

This is my SMTP config

![Schermata 2020-04-24 alle 18 06 38](https://user-images.githubusercontent.com/6501444/80233183-60f1f300-8656-11ea-8755-1f16e7b8313a.png)

And when I invite an user no email is sent.

This is the debug I get

```
[2020-04-24 18:09:28][request][INFO] GET /admin
[2020-04-24 18:09:28][response][INFO] GET /admin (admin_page) => 200 OK
[24/Apr/2020:18:09:28 +0200] 200 -(192.168.1.4) GET /admin HTTP/1.1 (Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.113 Safari/537.36)
[2020-04-24 18:09:29][request][INFO] GET /bwrs_static/bootstrap.css
[2020-04-24 18:09:29][response][INFO] GET /bwrs_static/<filename> (static_files) => 200 OK
[2020-04-24 18:09:29][request][INFO] GET /bwrs_static/md5.js
[2020-04-24 18:09:29][response][INFO] GET /bwrs_static/<filename> (static_files) => 200 OK
[2020-04-24 18:09:29][request][INFO] GET /bwrs_static/bootstrap-native-v4.js
[2020-04-24 18:09:29][response][INFO] GET /bwrs_static/<filename> (static_files) => 200 OK
[24/Apr/2020:18:09:29 +0200] 200 -(192.168.1.4) GET /bwrs_static/md5.js HTTP/1.1 (Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.113 Safari/537.36)
[24/Apr/2020:18:09:29 +0200] 200 -(192.168.1.4) GET /bwrs_static/bootstrap-native-v4.js HTTP/1.1 (Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.113 Safari/537.36)
[2020-04-24 18:09:30][request][INFO] GET /bwrs_static/identicon.js
[2020-04-24 18:09:30][response][INFO] GET /bwrs_static/<filename> (static_files) => 200 OK
[24/Apr/2020:18:09:30 +0200] 200 -(192.168.1.4) GET /bwrs_static/identicon.js HTTP/1.1 (Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.113 Safari/537.36)
[24/Apr/2020:18:09:30 +0200] 200 -(192.168.1.4) GET /bwrs_static/bootstrap.css HTTP/1.1 (Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.113 Safari/537.36)
[2020-04-24 18:09:30][request][INFO] GET /favicon.ico
[2020-04-24 18:09:30][response][INFO] GET /<p..> [10] (web_files) => 200 OK
[24/Apr/2020:18:09:30 +0200] 200 -(192.168.1.4) GET /favicon.ico HTTP/1.1 (Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.113 Safari/537.36)
[2020-04-24 18:09:43][request][INFO] POST /admin/test/smtp/
[2020-04-24 18:09:44][lettre::smtp::client][DEBUG] connecting to 173.194.76.109:465
[2020-04-24 18:09:54][lettre::smtp::client][DEBUG] Read: 
[2020-04-24 18:09:54][lettre::smtp::client][DEBUG] Wrote: QUIT<CRLF>
[2020-04-24 18:09:54][lettre::smtp::client][DEBUG] Read: 
[2020-04-24 18:09:54][error][ERROR] LetreErr.
[CAUSE] Parsing(
    Tag,
)
[2020-04-24 18:09:54][response][INFO] POST /admin/test/smtp (test_smtp) => 400 Bad Request
[24/Apr/2020:18:09:54 +0200] 400 -(192.168.1.4) POST /admin/test/smtp/ HTTP/1.1 (Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.113 Safari/537.36)
```

**Comment by franc6** [comment]
@woody4165 Port 465 is usually an encrypted tunnel.  I think you need to check the "Use explicit TLS" option.

**Comment by woody4165** [comment]
@franc6 on the Google IMAP port 465 is for SSL and 587 for TLS
![Schermata 2020-05-03 alle 07 59 51](https://user-images.githubusercontent.com/6501444/80907115-1e927b00-8d14-11ea-83e4-cc61643bcd93.png)

Thanks

**Comment by franc6** [comment]
Right. The terminology is unclear, largely because SSL and TLS are interchangeable. According to your documentation, port 587 accepts plaintext connections and supports the STARTTLS command (that’s usually how it’s set up), and port 465 requires an encrypted connection.  Given how I got the mail configuration working for port 587, I’m pretty sure that second option tells it to make an encrypted connection, while the first one indicates it should try to use the STARTTLS command. Hence, I suggested enabling the second one. :)
